ethcore/src/encoded.rs

```diff
@@ -205,7 +205,7 @@ impl Block {
 	pub fn header_view(&self) -> HeaderView { self.view().header_view() }
 
 	/// Decode to a full block.
-	pub fn decode(&self) -> FullBlock { ::rlp::decode(&self.0) }
+	pub fn decode(&self) -> FullBlock { ::rlp::decode(&self.0).unwrap_or(FullBlock::default()) }
```

Comment:
Same here. Returning `default` is not a good pattern and may lead too undefined behaviours

Reply:
https://github.com/paritytech/parity/issues/8553